### PR TITLE
Fix bug which deleted project bucket instead of project contents only

### PR DESCRIPTION
**Pull request by inaod568**
# Description

A user at NGI Stockholm noted that he created a project, uploaded data, used `dds data rm --rm-all` and then tried to upload data to the same project. The last step resulted in a long traceback complaining that there was no such bucket. Turns out the `delete_project_contents` used by the `RemoveContents` endpoint, and by the project-deleting/archiving also removes the buckets which is not intended. 

In this PR I have changed the name of the S3Connector method to `remove_bucket_contents` instead of `remove_bucket` and added a kwarg called `delete_bucket` (default False). When the projects are being archived or deleted, this should be set to True. 

- [x] Summary of the changes and the related issue
- [x] Relevant motivation and context
- [ ] Any dependencies that are required for this change

Fixes DDS-1283

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

Please delete options that are not relevant.

- [x] Any dependent changes have been merged and published in downstream modules
- [x] Rebase/merge the branch which this PR is made to
- [ ] Changes to the database schema: A new migration is included in the PR

## Formatting and documentation

- [x] I have added a row in the [changelog](../CHANGELOG.md)
- [x] The code follows the style guidelines of this project: Black / Prettier formatting
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Tests

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1172"></a>

### CODE REVIEW

File: dds_web/api/project.py [hunk 1]
```diff
@@ -545,12 +545,12 @@ def delete(self):
...
 
     @staticmethod
-    def delete_project_contents(project):
+    def delete_project_contents(project, delete_bucket=False):
         """Remove project contents"""
         # Delete from cloud
         with ApiS3Connector(project=project) as s3conn:
             try:
-                s3conn.remove_bucket()
+                bucket = s3conn.remove_bucket_contents(delete_bucket=delete_bucket)
```
Comment (valyo) [inline dds_web/api/project.py:553]: not sure I understand this: delete_project_contents does not seem to return anything and "bucket" is not used anywhere later on.

**Review by valyo — APPROVED** [review]
Worked when tested and seems fine, apart from my comment

File: dds_web/api/project.py (hunk 1 above)
Comment (inaod568) [reply dds_web/api/project.py:553]: Ah, this was something I was trying out and then forgot to remove it.

Comment (inaod568) [reply dds_web/api/project.py:553]:
```suggestion
                s3conn.remove_bucket_contents(delete_bucket=delete_bucket)
```

**Pull request merged**